JavaScript/JavaScriptSDK.Tests/Selenium/ai.tests.d.ts

```diff
@@ -2259,6 +2283,10 @@ declare module Microsoft.ApplicationInsights {
          * The object describing a session tracked by this object.
          */
         session: Context.Session;
+        /**
+         * AppId of this component if returned by the backend.
+         */
+        appId: () => string;
```

Comment:
Is this `TelemetryContext` object represented with the single ikey? In .NET we are keeping a hash map as every item may have it's own ikey.

Reply:
we don't support multi-key scenario in javascript sdk and, yes, we do have only one TelemetryContext for the whole execution.

Reply:
API reference says you can change [config properties dynamically](https://github.com/Microsoft/ApplicationInsights-JS/blob/master/API-reference.md#config). Also `ikey` is part of an [envelope](https://github.com/Microsoft/ApplicationInsights-JS/blob/master/API-reference.md#class-envelope) which is accessible from telemetry processor.

So it is not completely unsupported. 

I'd suggest to keep a hash map instead. It would be more error-prone.

Reply:
Fair. It's partially supported. Please create an issue about correct support for multi-ikey behavior in JS SDK.

Reply:
please create an issue or address it in PR now =)

Reply:
It's not about one line of change: Improve multi-ikey support #576